src/helpers/auth.js

```diff
@@ -0,0 +1,47 @@
+const fetch = require("node-fetch");
+const got = require('got');
+
+const AUTH_BASE_URL = "https://sigarra.up.pt/feup/pt/mob_val_geral.autentica?";
+
+
+// Called sigarra api to make authentication
+/*async function getCookie() {
+
+    // build URL with credentials
+    const urlRequest = `${AUTH_BASE_URL}pv_login=${encodeURIComponent(process.env.UP_USER)}&pv_password=${encodeURIComponent(process.env.UP_PWD)}`;
+
+    const res = await fetch(urlRequest, {
+        method: "POST"
+    });
+    if (res.ok) {
+        const parts = res.headers.get("set-cookie").split(";");
+        const SI_SESSION = parts[0];
+        const SI_SECURITY = parts[2].split(", ")[1];
+        cookie = [SI_SESSION, SI_SECURITY].join(";");
+        return cookie;
+    }
+    console.error(`Request Failed: ${res.status}-${res.statusText}`);
+    return null;
+}*/
+
+async function getCookie() {
+
+    // build URL with credentials
+    const urlRequest = `${AUTH_BASE_URL}pv_login=${encodeURIComponent(process.env.UP_USER)}&pv_password=${encodeURIComponent(process.env.UP_PWD)}`;
+
+    got.post(urlRequest)
+        .then((response) => {
+            const parts = response.headers["set-cookie"];
+            const SI_SESSION = parts[0].split(";")[0];
+            const SI_SECURITY = parts[1].split(";")[0];
+            cookie = [SI_SESSION, SI_SECURITY].join(";");
+            console.log(cookie);
+            return cookie;
```

Comment:
You are returning from the arrow function, not from 'getCookie' itself.
Try to use `const response = await got(url);`

Reply:
Understood